subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,305 @@
+		AdditionalNtpSource:      "test.ntp.source",
+		PullSecretRef:            secretRef,
+		ProvisionRequirements: v1alpha1.ProvisionRequirements{
+			ControlPlaneAgents: 3,
+			WorkerAgents:       0,
+			AgentSelector:      metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+		},
+	}
+}
+
+func compareSpecToCluster(cluster *common.Cluster, spec *v1alpha1.ClusterSpec) {
+	Expect(cluster.Name).Should(Equal(spec.Name))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.BaseDNSDomain).Should(Equal(spec.BaseDNSDomain))
+	Expect(cluster.ClusterNetworkCidr).Should(Equal(spec.ClusterNetworkCidr))
+	Expect(cluster.ClusterNetworkHostPrefix).Should(Equal(spec.ClusterNetworkHostPrefix))
+	Expect(cluster.ServiceNetworkCidr).Should(Equal(spec.ServiceNetworkCidr))
+	Expect(cluster.IngressVip).Should(Equal(spec.IngressVip))
+	Expect(cluster.SSHPublicKey).Should(Equal(spec.SSHPublicKey))
+	Expect(swag.BoolValue(cluster.VipDhcpAllocation)).Should(Equal(spec.VIPDhcpAllocation))
+	Expect(cluster.HTTPProxy).Should(Equal(spec.HTTPProxy))
+	Expect(cluster.HTTPSProxy).Should(Equal(spec.HTTPSProxy))
+	Expect(cluster.NoProxy).Should(Equal(spec.NoProxy))
+	Expect(swag.BoolValue(cluster.UserManagedNetworking)).Should(Equal(spec.UserManagedNetworking))
+	Expect(cluster.OpenshiftVersion).Should(Equal(spec.OpenshiftVersion))
+	Expect(cluster.AdditionalNtpSource).Should(Equal(spec.AdditionalNtpSource))
+	Expect(cluster.PullSecret).Should(Equal(pullSecret))
+}
+
+func deployImageCRD(ctx context.Context, client k8sclient.Client, spec *v1alpha1.ImageSpec) {
+	err := client.Create(ctx, &v1alpha1.Image{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Image",
+			APIVersion: getAPIVersion(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: Options.Namespace,
+			Name:      spec.ClusterRef.Name,
+		},
+		Spec: *spec,
+	})
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+func waitForImageCRDState(
+	ctx context.Context, client k8sclient.Client, key types.NamespacedName, message string, timeout int) {
+
+	image := &v1alpha1.Image{}
+	start := time.Now()
+	for time.Duration(timeout)*time.Second > time.Since(start) {
+		image = getImageCRD(ctx, client, key)
+		if image.Status.State != "" {
+			Expect(image.Status.State).Should(Equal(message))
+		}
+		time.Sleep(time.Second)
+	}
+	Expect(image.Status.State).Should(Equal(message))
+}
+
+func getImageCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Image {
+	image := &v1alpha1.Image{}
+	err := client.Get(ctx, key, image)
+	Expect(err).ShouldNot(HaveOccurred())
+	return image
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+}
+
+var _ = Describe("cluster installation", func() {
```

Comment:
lets mark all the kube-api tests somehow 
```suggestion
var _ = Describe("[kube-api] cluster installation", func() {
```